src/main/java/javaslang/collection/HashArrayMappedTrie.java

```diff
@@ -328,21 +328,21 @@ private IndexedNode(int bitmap, List<AbstractNode<K, V>> subNodes) {
             int bit = mask;
             int count = 0;
             List<AbstractNode<K, V>> sub = subNodes;
-            List<AbstractNode<K, V>> arr = List.empty();
+            final Object[] arr = new Object[BUCKET_SIZE];
```

Comment:
I don't like to pass a Class&lt;T&gt; as a parameter because it will change the Array's methods signatures as compared with all the other